docs/modules/clients/pages/java-thin-client.adoc

```diff
@@ -0,0 +1,1126 @@
+----
+
+[[configuring-client-connection-retry]]
+=== Configuring Client Connection Retry
+
+When the client is disconnected from the cluster or trying to connect to a one
+for the first time, it searches for new connections. You can configure the frequency
+of the connection attempts and client shutdown behavior using
+`ConnectionRetryConfig` (programmatically).
+
+[source,java]
+----
+ClientConfig config = new ClientConfig();
+ClientConnectionStrategyConfig connectionStrategyConfig = config.getConnectionStrategyConfig();
+ConnectionRetryConfig connectionRetryConfig = connectionStrategyConfig.getConnectionRetryConfig();
+connectionRetryConfig.setInitialBackoffMillis(1000)
+                     .setMaxBackoffMillis(60000)
+                     .setMultiplier(2)
+                     .setClusterConnectTimeoutMillis(50000)
+                     .setJitter(0.2);
+
+----
+
+The following are configuration element descriptions:
+
+* `initial-backoff-millis`: Specifies how long to wait (backoff), in milliseconds, after the first failure before retrying.
+Its default value is 1000 ms.
+* `max-backoff-millis`: Specifies the upper limit for the backoff in milliseconds.
+Its default value is 30000 ms.
+* `multiplier`: Factor to multiply the backoff after a failed retry.
+Its default value is 1.05.
+* `cluster-connect-timeout-millis`: Timeout value in milliseconds for the client to give up
+to connect to the current cluster. Its default value is `-1`, i.e., infinite.
+For the default value, the client will not stop trying to
+connect to the target cluster (infinite timeout). If the failover client is used
+with the default value of this configuration element, the failover client will try
+to connect alternative clusters after 120000 ms (2 minutes). For any other value,
+both the client and the failover client will use this as it is.
+* `jitter`: Specifies by how much to randomize backoffs. Its default value is 0.
+
+A pseudo-code is as follows:
+
+[source,java]
+----
+ begin_time = getCurrentTime()
+ current_backoff_millis = INITIAL_BACKOFF_MILLIS
+ while (TryConnect(connectionTimeout)) != SUCCESS) {
+    if (getCurrentTime() - begin_time >= CLUSTER_CONNECT_TIMEOUT_MILLIS) {
+         // Give up to connecting to the current cluster and switch to another if exists.
+         // For the default values, CLUSTER_CONNECT_TIMEOUT_MILLIS is infinite for the
+         // client and equal to the 120000 ms (2 minutes) for the failover client.
+    }
+    Sleep(current_backoff_millis + UniformRandom(-JITTER * current_backoff_millis, JITTER * current_backoff_millis))
+    current_backoff = Min(current_backoff_millis * MULTIPLIER, MAX_BACKOFF_MILLIS)
+}
+----
+
+Note that, `TryConnect` above tries to connect to any member that the client knows,
+and for each connection we have a connection timeout; see the
+<<setting-connection-timeout, Setting Connection Timeout section>>.
+
+[[blue-green-deployment-and-disaster-recovery]]
+== Blue-Green Deployment
```

Comment:
Why is "Ordering of Clusters When Clients Try to Connect" section missing from thin client? See https://github.com/hazelcast/hz-docs/blob/main/docs/modules/clients/pages/java.adoc

Reply:
Should this section be added back in then?

Reply:
Yes, I don't see the reason of excluding this section. The feature might be missing in the thin client or it can be forgotten (or the source page was old compared to main branch)

Reply:
Alexey will look at this section soon

Reply:
Added the section (allow some time to address other changes and commit the changes)